eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -0,0 +1,194 @@
+  TRC (..),
+  TransitionRule,
+  judgmentContext,
+  trans,
+ )
+import Data.Default.Class (Default)
+import qualified Data.Map.Strict as Map
+import GHC.Records (HasField)
+
+data ConwayEpochEvent era
+  = PoolReapEvent (Event (EraRule "POOLREAP" era))
+  | SnapEvent (Event (EraRule "SNAP" era))
+
+instance
+  ( EraTxOut era
+  , HasField "_keyDeposit" (PParams era) Coin
+  , HasField "_poolDeposit" (PParams era) Coin
+  , Embed (EraRule "SNAP" era) (ConwayEPOCH era)
+  , Environment (EraRule "SNAP" era) ~ LedgerState era
+  , State (EraRule "SNAP" era) ~ SnapShots (EraCrypto era)
+  , Signal (EraRule "SNAP" era) ~ ()
+  , Embed (EraRule "POOLREAP" era) (ConwayEPOCH era)
+  , Environment (EraRule "POOLREAP" era) ~ PParams era
+  , State (EraRule "POOLREAP" era) ~ ShelleyPoolreapState era
+  , Signal (EraRule "POOLREAP" era) ~ EpochNo
+  , Default (PParams era)
+  , Eq (UpecPredFailure era)
+  , Show (UpecPredFailure era)
+  , Default (PPUPState era)
+  , EraTallyState era
+  ) =>
+  STS (ConwayEPOCH era)
+  where
+  type State (ConwayEPOCH era) = EpochState era
+  type Signal (ConwayEPOCH era) = EpochNo
+  type Environment (ConwayEPOCH era) = ()
+  type BaseM (ConwayEPOCH era) = ShelleyBase
+  type PredicateFailure (ConwayEPOCH era) = ShelleyEpochPredFailure era
+  type Event (ConwayEPOCH era) = ConwayEpochEvent era
+  transitionRules = [epochTransition]
+
+epochTransition ::
+  forall era.
+  ( Embed (EraRule "SNAP" era) (ConwayEPOCH era)
+  , Environment (EraRule "SNAP" era) ~ LedgerState era
+  , State (EraRule "SNAP" era) ~ SnapShots (EraCrypto era)
+  , Signal (EraRule "SNAP" era) ~ ()
+  , Embed (EraRule "POOLREAP" era) (ConwayEPOCH era)
+  , Environment (EraRule "POOLREAP" era) ~ PParams era
+  , State (EraRule "POOLREAP" era) ~ ShelleyPoolreapState era
+  , Signal (EraRule "POOLREAP" era) ~ EpochNo
+  ) =>
+  TransitionRule (ConwayEPOCH era)
+epochTransition = do
+  TRC
+    ( _
+      , EpochState
+          { esAccountState = acnt
+          , esSnapshots = ss
+          , esLState = ls
+          , esPrevPp = pr
+          , esPp = pp
+          , esNonMyopic = nm
+          }
+      , e
+      ) <-
+    judgmentContext
+  let utxoSt = lsUTxOState ls
+  let DPState dstate pstate = lsDPState ls
+  ss' <-
+    trans @(EraRule "SNAP" era) $ TRC (ls, ss, ())
+
+  let PState pParams fPParams _ _ = pstate
+      ppp = eval (pParams ⨃ fPParams)
+      pstate' =
+        pstate
+          { psStakePoolParams = ppp
+          , psFutureStakePoolParams = Map.empty
+          }
+  PoolreapState utxoSt' acnt' dstate' pstate'' <-
+    trans @(EraRule "POOLREAP" era) $
+      TRC (pp, PoolreapState utxoSt acnt dstate pstate', e)
+
+  let adjustedDPstate = DPState dstate' pstate''
+      epochState' =
+        EpochState
+          acnt'
+          ss'
+          (ls {lsUTxOState = utxoSt', lsDPState = adjustedDPstate})
+          pr
+          pp
+          nm
+
+  let
+    -- At the epoch boundary refunds are made, so we need to change what
```

Comment:
I suspect that this is a comment that you inherited, but it is confusing me and I'd like to take the opportunity to make it better. Before the individual deposit tracking change, we were indeed giving refunds on the epoch boundary, but I can't see how this is true anymore.

I think `oblgNew` might also needs a bit more clarification. I think we are only using this `sutxosDeposited` value as a sanity check anymore (which is really nice in property testing). If so, we can mention this fact. @TimSheard would probably remember best.

Reply:
Tim thought I should just get rid of it, so I'm gonna remove this.